src/index.js

```diff
@@ -11,26 +10,12 @@ const run = async () => {
   core.info(`Running for current SHA ${sha}`)
   
   try {
-    // These are the string names of the owner and repo
-    const { owner, repo } = github.context.repo
-
-    const token = core.getInput('token', { required: true })
-    const octokit = github.getOctokit(token)
-
-    // Get the repo object, which contains the `default_branch` property.
-    const repoResult = await octokit.request('GET /repos/{owner}/{repo}', { owner, repo })
-    const defaultBranch = repoResult.data.default_branch
-    core.info(`Using default branch '${defaultBranch}'`)
-
-    // Fetch the current SHA of the default branch
-    const ref = `heads/${defaultBranch}`
-    const refResult = await octokit.rest.git.getRef({ owner, repo, ref })
-    core.info(`Found '${defaultBranch}' with SHA ${refResult.data.object.sha}`)
-
     // inputResult will be 'unknown' if we're in "restore only" mode.
     const inputResult = core.getInput('result')
-    const key = 'cache-result-action-' + sha + '-' + Math.floor(Date.now() / 1000)
-    const cacheKey = await cache.restoreCache([RESULT_PATH], key, ['cache-result-action-' + sha])
+    const cacheGroup = core.getInput('cache-group')
+    const key = ['cache-result-action', cacheGroup, sha, Math.floor(Date.now() / 1000)].filter(Boolean).join('-')
+
+    await cache.restoreCache([RESULT_PATH], key, ['cache-result-action-' + sha])
```

Comment:
The third argument to `cache.restoreCache` should have the full cache path before the timestamp:

```suggestion
    const keyPrefix = `cache-result-action-${cacheGroup}-${sha}`
    const key = keyPrefix + '-' + Math.floor(Date.now() / 1000)

    await cache.restoreCache([RESULT_PATH], key, [keyPrefix])
```